general/mem_manager.hpp

```diff
@@ -110,13 +110,28 @@ MemoryClass operator*(MemoryClass mc1, MemoryClass mc2);
 /** The template class parameter, T, must be a plain-old-data (POD) type.
 
     In many respects this class behaves like a pointer:
-    * When destroyed, a Memory object does NOT automatically delete any
+    - When destroyed, a Memory object does NOT automatically delete any
       allocated memory.
-    * Only the method Delete() will deallocate a Memory object.
-    * Other methods that modify the object (e.g. New(), Wrap(), etc) will simply
-      overwrite the old contents.
-    * One difference with a pointer is that a const Memory object does not allow
-      modification of the content (unlike e.g. a const pointer).
+    - Only the method Delete() will deallocate a Memory object.
+    - Other methods that modify the object (e.g. New(), Wrap(), etc) will
+      simply overwrite the old contents.
+    In other aspects this class differs from a pointer:
+    - Pointer arithmetic is not supported, MakeAlias() should be used instead.
+    - Const Memory object does not allow modification of the content
+      (unlike e.g. a const pointer).
+    - Move constructor and assignement will transfer ownership flags, and
+      Reset() the moved Memory object.
+    - Copy constructor and assignement copy flags. This may result in two Memory
+      objects owning the data which is an invalid state. This invalid state MUST
+      be resolved by users manually using SetHostPtrOwner(),
+      SetDevicePtrOwner(), or ClearOwnerFlags(). One can also just NOT use
+      Delete() on one of the two Memory objects, however this is discouraged as
+      it is counterproductive to have developpers be responsible for tracking
+      ownership when flags are dedicated to this task.
```

Comment:
Can you clarify this sentence? I can't understand it as written

"One can also just NOT use Delete() on one of the two Memory objects, however this is discouraged as it is counterproductive to have developpers be responsible for tracking ownership when flags are dedicated to this task."

Also, I will replace "as it is counterproductive to have developers be responsible for tracking ownership when flags are dedicated to this task" with "since ownership is tracked by internal flags".

Reply:
done in [b3a1202](https://github.com/mfem/mfem/pull/2355/commits/b3a120277a1fb9e083cc06f651021294fa8d76b6)